client.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"net/http"
+	"net/http/httptrace"
+	"net/http/httputil"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+)
+
+// NOTE: Once we refactor the juju tests enough that they do not use
+// a RoundTripper on the DefaultTransport, NewClient can always return
+// a Client with a locally constructed Transport via NewHttpTLSTransport
+// and init() will no longer be needed.
+func init() {
+	defaultTransport := http.DefaultTransport.(*http.Transport)
```

Comment:
Can we remove this `init()` function, it's considered an anti-pattern?

Also, this modifies every `http.DefaultTransport` and I'm pretty sure we don't want to do that...!

Reply:
Also, nobody is using `defaultTransport` locally.

Reply:
defaultTransport is used for an argument with the function calls after.  Unfortunately we need this to get some transport changes in for juju.  juju tests currently require being able to change the DefaultTransport to add different schemes for test and replace the RoundTripper.  Updating the tests needs to be done, but make take weeks of effort which we do not have at this time.  Small updates can be make to use httptest instead and chip away at the problem.  The follow on pr in juju/juju starts that.